### PR TITLE
Java Arraylist replaceAll()

### LINKED ISSUE #4345: [Term Entry] Java ArrayList .replaceAll()
### Reason/inspiration (optional)

Please add the `.replaceAll()` method as a new term entry for [ArrayList](https://www.codecademy.com/resources/docs/java/array-list)�. Provide the entry in a markdown file and place under the file path `content/java/concepts/array-list/terms/replaceAll/replaceAll.md`

The entry should have the following:
�	An introductory paragraph about what the method is and does.
�	A syntax section with a pseudo code block with an explanation of it.
�	An example showing the method in use.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by CodedByCoffee**
### Description

Added documentation for the replaceAll() method in the ArrayList class as an .md file called replaceAll.md

### Issue Solved

Closes #4345

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by cigar-galaxy82** [comment]
Hi, @CodedByCoffee can you please sign the CLA.

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+Title: '.replaceAll()'
+Description: 'Replaces all occurrences of a given value ArrayList.'
```
Comment (cigar-galaxy82) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:3]:
```suggestion
Description: 'Replaces all occurrences of a given value in an ArrayList.'
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  - 'An nth subject name'
+Tags:
+  - 'Computer Science'
+  - 'Information Technology'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.replaceAll()`** method of the `ArrayList` class takes an Operator as argument and applies it to every element in the List
```
Comment (cigar-galaxy82) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:16]:
```suggestion
The `.replaceAll()` method of the `ArrayList` class takes an operator as argument and applies it to every element in the list
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+---
+Title: '.replaceAll()'
+Description: 'Replaces all occurrences of a given value ArrayList.'
+Subjects:
+  - 'A subject name'
+  - 'A second subject name'
+  - 'An nth subject name'
```
Comment (cigar-galaxy82) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:7]: Two Subjects works for this entry

**Comment by cigar-galaxy82** [comment]
Hey @CodedByCoffee 

Thank you for contributing to Codecademy Docs! The entry is well written!

I've suggested a few changes modify them at your convenience

Thank you!

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md (hunk 2 above)
Comment (CodedByCoffee) [reply content/java/concepts/array-list/terms/replaceAll/replaceAll.md:16]: Should we not keep the first replaceAll() occurence bold as per this guide? https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#writing-entries.
But maybe remove the backtick?

**Comment by CodedByCoffee** [comment]
> Hi, @CodedByCoffee can you please sign the CLA.

I cannot find how I do that, where is the CLA and how do I sign it? 😅

**Comment by cigar-galaxy82** [comment]
Hi @CodedByCoffee don't worry about the CLA just make the changes sometimes our CLAassistant behave unpredictably

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md (hunk 2 above)
Comment (CodedByCoffee) [reply content/java/concepts/array-list/terms/replaceAll/replaceAll.md:15]: @cigar-galaxy82 See my questsion above ☝️

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Arrays'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:13]:
```suggestion
---
Title: '.replaceAll()'
Description: 'Replaces all occurrences of elements in the ArrayList with a specified new value..'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Data Types'
  - 'Java'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+The **`.replaceAll()`** method of the `ArrayList` class takes an Operator as argument and applies it to every element in the List
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:15]:
```suggestion
The **`.replaceAll()`** method in Java is used for transforming all elements of a list according to a specified unary operation, effectively updating each element in place with the result of the operation applied to it.
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## Syntax
+
+Applies `operator` to each element in ArrayList `myArrayList` and replaces the element with a new value
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:19]:
```suggestion
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+```pseudo
+myArrayList.replaceAll(operator);
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:21]:
```suggestion
arrayList.replaceAll(UnaryOperator<E> operator)
```

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+```
+
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:23]: Please describe the parameters in the following format:

- `arrayList`: parameter description
- `UnaryOperator<E> operator`: parameter description

File: content/java/concepts/array-list/terms/replaceAll/replaceAll.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+}
+```
+
+The output for this code would look like this:
+
+```shell
+Cities in uppercase: 
+[STOCKHOLM, LONDON, PARIS, BERLIN]
+Updated cities in lowercase: 
+[stockholm, london, paris, berlin]
```
Comment (mamtawardhani) [inline content/java/concepts/array-list/terms/replaceAll/replaceAll.md:55]:
```suggestion
Cities in uppercase: [STOCKHOLM, LONDON, PARIS, BERLIN]
Updated cities in lowercase: [stockholm, london, paris, berlin]
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @CodeByCoffee! Thank you for the contribution to Codecademy Docs! 😄 

I've requested some changes for this entry, could you please review and modify those at your earliest convenience? Thank you!

**Comment by mamtawardhani** [comment]
Hey, @CodedByCoffee could you make the requested changes at your earliest convenience, so that this PR can be merged soon? Thanks 😄

**Comment by mamtawardhani** [comment]
> [CLA assistant check](https://cla-assistant.io/Codecademy/docs?pullRequest=4642) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=4642) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ mohitO12-dev❌ GROUPINFRA\mattias.gradin
> 
> **GROUPINFRA\mattias.gradin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=4642) it.

@CodedByCoffee here is the CLA, could you please check if you can sign it? Let me know 😃

**Comment by CodedByCoffee** [comment]
@mamtawardhani So now I have fixed all that you requested, please have a look when you have the time :)

**Comment by mamtawardhani** [comment]
> @mamtawardhani So now I have fixed all that you requested, please have a look when you have the time :)

Hey @CodedByCoffee! 
 As it seems, all the commits are made by `GROUPINFRA\mattias.gradin` and none by `CodedByCoffee`. Hence, the CLA only wants the signature from `GROUPINFRA\mattias.gradin`. This most probably means that the commits are made locally by `GROUPINFRA\mattias.gradin` and the email address used for making them is not added to the `CodedByCoffee`GitHub account.
 
I would suggest you add the email address to your github account and also suggest visiting the `add the email address used for this commit to your account link in the CLAassistant` message that explains the process of doing this.

Please refer to this: [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user). and add the correct email id that was used to make the commits which is `GROUPINFRA\mattias.gradin``

**Comment by CodedByCoffee** [comment]
> > @mamtawardhani So now I have fixed all that you requested, please have a look when you have the time :)
> 
> Hey @CodedByCoffee! As it seems, all the commits are made by `GROUPINFRA\mattias.gradin` and none by `CodedByCoffee`. Hence, the CLA only wants the signature from `GROUPINFRA\mattias.gradin`. This most probably means that the commits are made locally by `GROUPINFRA\mattias.gradin` and the email address used for making them is not added to the `CodedByCoffee`GitHub account.
> 
> I would suggest you add the email address to your github account and also suggest visiting the `add the email address used for this commit to your account link in the CLAassistant` message that explains the process of doing this.
> 
> Please refer to this: [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user). and add the correct email id that was used to make the commits which is `GROUPINFRA\mattias.gradin`

@mamtawardhani Now I have added my work email to my Github account and it seems the CLA is recognizing that I have done so 👍

**Comment by mamtawardhani** [comment]
@cigar-galaxy82 please approve this PR.

**Review by mamtawardhani — APPROVED** [review]
Thank you @CodedByCoffee for thie entry! 😃 

It is ready to be merged! 🚀

**Comment by CodedByCoffee** [comment]
> Thank you @CodedByCoffee for thie entry! 😃
> 
> It is ready to be merged! 🚀

@mamtawardhani Great 😃

**Review by cigar-galaxy82 — APPROVED** [review]

**Pull request merged**